app/controllers/projects_controller.rb

```diff
@@ -63,20 +63,28 @@ def index
                      current_power.samples
                    end
 
-        @projects = if only_library || exclude_library
-                      @samples.group(:project).count
+        # Retrieve a json of projects associated with samples;
+        # augment with number_of_samples, hosts, tissues.
+        # TODO: Why are we passing host / tissue for each individual sample, with a lot of duplication?
+        #       Can't we tally the hosts and tissues up on the server side, since only summary counts are shown on the page?
+        @projects = if only_updatable
```

Comment:
Just notice this ignores `only_library` and `exclude_library`, which means that if any of those are set will return the wrong list of projects.

As a side note, In my current PR these variables change to domain that can have the values "library", "public" and "updatable". Default, for now, is all the projects user can see.

Reply:
Actually further down it only keeps the intersection of `@projects` (all projects the user has access to) and `@samples` (only the samples that are in the specified option: `only_library` or `exclude_library`)